src/routines/write_model.jl

```diff
@@ -1,70 +1,75 @@
+struct OptimizerLog
+
+    termination_status::Enum
+    primal_status::Enum
+    dual_status::Enum
+    solve_time::Float64
+
+end
+
+
 # taking the outputted files for the variable DataFrame and writing them to a featherfile
-function _write_variable_results(vars_results::Dict{Symbol, DataFrames.DataFrame}, save_path::AbstractString)
+function write_data(vars_results::Dict{Symbol, DataFrames.DataFrame}, save_path::AbstractString; kwargs...)
 
+    file_type = get(kwargs, :file_type, Feather)
     for (k,v) in vars_results
-         file_path = joinpath(save_path,"$(k).feather")
-         Feather.write(file_path, vars_results[k])
+         file_path = joinpath(save_path,"$(k).$file_type")
+         file_type.write(file_path, vars_results[k])
     end
 
     return
-
 end
 
-function _write_variable_results(canonical::CanonicalModel, save_path::AbstractString)
 
-    for (k,v) in vars(canonical)
-         file_path = joinpath(save_path,"$(k).feather")
-         Feather.write(file_path, _result_dataframe_vars(v))
+function write_data(vars_results::OperationModel, save_path::AbstractString; kwargs...)
+
+    file_type = get(kwargs, :file_type, Feather)
+    for (k,v) in vars(vars_results.canonical)
+         file_path = joinpath(save_path,"$(k).$file_type")
+         file_type.write(file_path, _result_dataframe_vars(v))
```

Comment:
I think it's unlikely that other formats will support the exact same write function as Feather.  And there is no error checking.  I suggest something like this instead:
`if file_type == Feather
    Feather.write(...)
else
    error("unsupported file type:  $file_type")
end

Another alternative is to discern the file type by looking at the file extension.  That is what I did here:  https://github.nrel.gov/Hosting-Capacity-Analysis/disco/blob/master/disco/utils/postprocess.py#L173, but I am sure that some people would not like that approach.

Reply:
If another developer read the function signature only then they might assume that they should pass "out.feather" instead of "out".  You could fix that with a docstring.  Or you could strip the extension if they pass one.

Reply:
I'm slightly confused, Wouldn't that if/else only support feather in that case? I see your point, but what about, say, CSV that is supported in this format. Should I do a try and catch statement? @daniel-thom

Reply:
Right, you would have to add an ```elseif CSV``` code block.